backend/backend.go

```diff
@@ -13,22 +13,31 @@ var (
 // A Backend is used to fetch values from a given key.
 type Backend interface {
 	Get(ctx context.Context, key string) ([]byte, error)
+	// String returns the name of the backend for comparing with the backend tag value.
+	String() string
```

Comment:
I think `String` is usually used to get a string representation of a type and not as an identifier used for comparison.
IMO `Name` seems more accurate in our case.